odxtools/odxlink.py

```diff
@@ -4,29 +4,14 @@
 from typing import Any, Dict, Iterable, List, Optional, Type, TypeVar, overload
 from xml.etree import ElementTree
 
-from .exceptions import OdxWarning, odxassert, odxraise
+from .exceptions import OdxWarning, odxassert, odxraise, odxrequire
 from .nameditemlist import OdxNamed, TNamed
 
 
 @dataclass(frozen=True)
 class OdxDocFragment:
```

Comment:
NamedTuple is better if you want to use this as dict key

Reply:
Hm, I'm not so sure: `NamedTuple` seems to be slightly faster and come with a slightly lower memory footprint but it severely limits flexibility because AFAICS there is no possibility to add custom methods like `from_et()`, i.e., it is not feasible to use it for e.g., `OdxLinkId` and `OdxLinkRef` which would IMO benefit quite a bit more from making them `NamedTuple`s than `OdxDocFragment`. That said, maybe I just do not understand `NamedTuple` fully yet?

Reply:
after a bit of googling, it turns out that it is possible to [create `NamedTuples` with user defined methods](https://stackoverflow.com/questions/3223236/creating-a-namedtuple-with-a-custom-hash-function). That said, inheritance is [quite akward](https://stackoverflow.com/questions/42385916/inheriting-from-a-namedtuple-base-class) for named tuples and -- more importantly -- requires quite a bit of manual code because the constructors of child classes must be aware of *all* constructor arguments required by the base class, i.e., I have not found a way of how an analog to `dataclass_fields_asdict()` can be implemented. (My thinking is that if we were to go the `NamedTuple` route thoroughly, we would need to subclass `OdxLinkRef` for e.g. `DynEndDopRef` and `RelatedDiagCommRef`. And if we are not doing this consistently, it is IMO better not to take this route at all...)